### PR TITLE
Add a new ignores flag to parse ignore patterns as array

### PR DESCRIPTION
**Pull request by muratcelep**
Q | A
-- | --
🐛 Bug fix? | no
🚀 New feature? | yes
⚠ Deprecations? | no
❌ BC Break | no
🔗 Related issues | 
❓ Documentation | TODO

## Description
This feature is for adding a new ignore flag to parse ignore patterns as arguments and as an alternative way to using driftignore file.

In order to avoid conflicts between ignore files and individual ignore patterns, whenever ignore patterns are present ignorefiles will not be processed.

### CODE REVIEW

**Comment by wbeuil** [comment]
I tested it on my side and it seems to work as intended.

I just find it really weird to pass ignores like that. A small example like this one `driftctl scan --ignores '#,*' --ignores '!aws_s3*'` and the result is to ignore everything (because of the wildcard) but aws_s3 resources, which would not have been the same with a simple driftignore:

```
#,*
!aws_s3*
```

It's only because of the way we pass the args in the CLI. Just my two cents here.

**Comment by muratcelep** [comment]
Right, the 'weird' behavior is because of the _StringSlice_ element which ends up creating three strings `'#',` `'*'` , `'!aws_s3*'` instead of 2 as in your .driftignore example. 

Using _StringArray_ would fix that, we would need to pass each ignorePattern with a separate --ignore param in that case which also should be doable.

File: pkg/filter/driftignore_test.go [hunk 1]
```diff
@@ -534,3 +534,51 @@ func TestDriftIgnore_IsTypeIgnored(t *testing.T) {
 		})
 	}
 }
+
+func TestDriftIgnore_IsTypeIgnoredWhenUsingIgnoreParams(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (muratcelep) [reply pkg/filter/driftignore_test.go]: @wbeuil merged the tests for ignore patterns into existing ones.

**Review by wbeuil — APPROVED** [review]
LGTM

**Pull request merged**